examples/general_demo.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+""" This script will:
+- download unsorted dicoms
+- run dcm2bids to convert to nifti with bids structure
+- unwrap phase
+- save wrapped and unwrapped plot of first X,Y volume as unwrap_phase_plot.png in the output directory
+"""
+
+import os
+import numpy as np
+from matplotlib.figure import Figure
+import glob
+import logging
+
+import nibabel as nib
+
+from shimmingtoolbox.unwrap import prelude
+from shimmingtoolbox.utils import run_subprocess
+from shimmingtoolbox import __dir_testing__
+from shimmingtoolbox import dicom_to_nifti
+
+
+def general_demo(path_output=os.path.join(os.path.curdir, 'output_dir')):
+    """
+    Args:
+        path_output (str): Output directory to store data and results.
+
+    Returns:
+        str: File name and path of output figure
+    """
+
+    logging.basicConfig(level='INFO')
+
+    # Download example data
+    path_testing_data = os.path.join(path_output, __dir_testing__)
+    run_subprocess('st_download_data {} --output {}'.format(__dir_testing__, path_testing_data))
+
+    # Transfer from dicom to nifti
+    path_dicom_unsorted = os.path.join(path_testing_data, 'dicom_unsorted')
+    path_nifti = os.path.join(path_output, 'niftis')
+    dicom_to_nifti(path_dicom_unsorted, path_nifti, subject_id='sub-01')
+
+    # Open phase data
+    fname_phases = glob.glob(os.path.join(path_nifti, 'sub-01', 'fmap', '*phase*.nii.gz'))
+
+    nii_phase_e1 = nib.load(fname_phases[0])
+    nii_phase_e2 = nib.load(fname_phases[1])
+
+    # Scale phase to radians
+    phase_e1 = np.interp(nii_phase_e1.get_fdata(), [0, 4096], [-np.pi, np.pi])
+    phase_e2 = np.interp(nii_phase_e2.get_fdata(), [0, 4096], [-np.pi, np.pi])
+
+    # Open mag data
+    fname_mags = glob.glob(os.path.join(path_nifti, 'sub-01', 'fmap', '*magnitude*.nii.gz'))
+
+    nii_mag_e1 = nib.load(fname_mags[0])
+    nii_mag_e2 = nib.load(fname_mags[1])
+
+    # TODO: create mask
+    # Call SCT or user defined mask
+    # mask = np.ones(phase_e1.shape)
```

Comment:
Ideally TODOs shouldn't make it into production – would this belong in an issue ticket instead?

Reply:
The issue #47 is already created, I could remove the TODOs in the script.